pkg/service/model_build_load_balancer.go

```diff
@@ -67,13 +78,85 @@ func (t *defaultModelBuildTask) buildLoadBalancerSpec(ctx context.Context, schem
 		Type:                   elbv2model.LoadBalancerTypeNetwork,
 		Scheme:                 &scheme,
 		IPAddressType:          &ipAddressType,
+		SecurityGroups:         securityGropus,
 		SubnetMappings:         subnetMappings,
 		LoadBalancerAttributes: lbAttributes,
 		Tags:                   tags,
 	}
 	return spec, nil
 }
 
+func (t *defaultModelBuildTask) buildLoadBalancerSecurityGroups(ctx context.Context, existingLB *elbv2deploy.LoadBalancerWithTags,
+	ipAddressType elbv2model.IPAddressType) ([]core.StringToken, error) {
+	if existingLB != nil && len(existingLB.LoadBalancer.SecurityGroups) == 0 {
+		return nil, nil
```

Comment:
Should it not be possible to add a SG to an existing NLB by explicitly setting the relevant annotation?